src/pages/ServiceDetails/ServiceInfo/ServiceInfoVirtualServices.tsx

```diff
@@ -85,6 +85,14 @@ class ServiceInfoVirtualServices extends React.Component<ServiceInfoVirtualServi
     };
   }
 
+  hasValidations(virtualService: VirtualService): boolean {
+    // This is insane, but doing return to the clause inside the if will cause compiler failure
+    if (this.props.validations && this.props.validations[virtualService.metadata.name]) {
```

Comment:
yep, I guess this is because `this.props.validations['vs-name']` returns a non-boolean value.
If you wanted to have it in one line, I guess it would be enough with addint the `!!` operator that converts whatever variable to a boolean. So this should work:
`return this.props.validations && !!this.props.validations[virtualService.metadata.name];`
Same thing would apply in the file right above: `ServiceInfoDestinationRules.tsx`